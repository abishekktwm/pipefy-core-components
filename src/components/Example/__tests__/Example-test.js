import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import Example from '../Example'

const props = {
  onClick: () => {}
}

it('maches the snapshot', () => {
  const tree = renderer.create(<Example {...props}>Facebook</Example>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('calls onClick on item click', function() {
  const onClick = jest.fn()
  const wrapper = mount(<Example {...props} onClick={onClick}> Aqui! </Example>)

  expect(onClick).not.toBeCalled()

  expect(wrapper.find('button').length).toEqual(1);
  wrapper.find('button').simulate('click')

  expect(onClick).toBeCalled()
})
