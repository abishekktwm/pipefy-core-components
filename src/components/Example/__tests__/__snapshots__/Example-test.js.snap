// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maches the snapshot 1`] = `
<button
  onClick={[Function]}
>
  2-Button: 
  Facebook
</button>
`;
